### PR TITLE
Cli argument parsing

### PR DESCRIPTION
**Pull request by Hackder**
- add cli argument parser

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    raise ValueError(f"Unknown argument: {name}")
+
+
+def parse_cli_params(argv: list[str]) -> CliOptions:
+    options = CliOptions()
+
+    if len(argv) <= 1:
+        return options
+
+    parsed_param = False
```
Comment (Vl4dk0) [inline cli.py:53]: what was parsed_param created for ? By the looks of it, it is not needed

File: cli.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+This is a simple CLI utility for submitting solution to L.I.S.T. automatically.
+It will take all of your project files, zip them acording to the L.I.S.T. requirements,
+submit them to the L.I.S.T. server, and run the tests for you. All automatically.
+"""
+
+
+def format_argument(arg: CliArgument) -> str:
+    aliases = [f"-{alias}" for alias in arg.aliases]
+    aliases = ", ".join(aliases)
+    return out.secondary(f"  --{arg.name}, {aliases}: ") + arg.description
```
Comment (Vl4dk0) [inline cli.py:98]: Let's not use secondary(blue), rather yellow.

File: cli.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            switch = arg[2:]
+        elif arg.startswith("-"):
+            switch = arg[1:]
+        elif not parsed_param:
+            parsed_param = True
+            setattr(options, cli_param.name, cli_param.value_parser(arg))
+            i += 1
+            continue
+        else:
+            raise ValueError(f"Unknown argument: {arg}")
```
Comment (Vl4dk0) [inline cli.py:68]: What is parsed_param for ? It feels meaningless.

Comment (Vl4dk0) [reply cli.py:68]: sorrko, ze to na tom preview nevidno, neviem ako spravit aby sa koment priradil k celej mnou oznacenej casti.

Comment (Hackder) [reply cli.py:68]: It is used to parse the param only once

**Pull request merged**